modules/distribution/src/repository/resources/conf/templates/repository/conf/identity/identity.xml.j2

```diff
@@ -276,63 +349,55 @@
             </UserConsentEnabledGrantType>
-        {% endfor %}
-        {% for grant_type in oauth.consent_enabled_grant_types %}
             <UserConsentEnabledGrantType>
-                <GrantTypeName>{{grant_type}}</GrantTypeName>
+                <GrantTypeName>implicit</GrantTypeName>
             </UserConsentEnabledGrantType>
-        {% endfor %}
         </UserConsentEnabledGrantTypes>
 
         <OAuthCallbackHandlers>
-        {% for callback_handler in oauth.callback_handler %}
+        {% for callback_handler in oauth.extensions.callback_handlers %}
             <OAuthCallbackHandler Class="{{callback_handler}}"/>
         {% endfor %}
         </OAuthCallbackHandlers>
 
         <TokenValidators>
-        {% for token_validator in oauth.default_token_validator %}
-            <TokenValidator type="{{token_validator.name}}" class="{{token_validator.class}}" />
-        {% endfor %}
-        {% for token_validator in oauth.token_validator %}
+        {% if oauth.token_validator.bearer.enable is sameas true %}
+            <TokenValidator type="bearer" class="{{oauth.token_validator.bearer.class}}"/>
+        {% endif %}
+        {% if oauth.token_validator.jwt.enable is sameas true %}
+            <TokenValidator type="jwt" class="{{oauth.token_validator.jwt.class}}"/>
+        {% endif %}
+        {% for token_validator in oauth.custom_token_validator %}
             <TokenValidator type="{{token_validator.type}}" class="{{token_validator.class}}" />
         {% endfor %}
         </TokenValidators>
 
         <!-- Scope validators list. The validators registered here wil be executed during token validation. -->
         <ScopeValidators>
-        {% for scope_validator in oauth.default_scope_validator %}
+        {% if oauth.scope_validator.jdbc.enable is sameas true %}
+            <ScopeValidator class="{{oauth.scope_validator.jdbc.class}}" />
+        {% endif %}
+        {% if oauth.scope_validator.xacml.enable is sameas true %}
+            <ScopeValidator class="{{oauth.scope_validator.xacml.class}}"/>
+        {% endif %}
+        {% for scope_validator in oauth.custom_scope_validator %}
             <ScopeValidator class="{{scope_validator.class}}" />
         {% endfor %}
-        {% for scope_validator in oauth.scope_validators %}
-            <ScopeValidator class="{{scope_validator}}" />
-        {% endfor %}
         </ScopeValidators>
 
         <!-- Scope handlers list. The handlers registered here will be executed at the scope validation phase while
              issuing access tokens. -->
         <ScopeHandlers>
-        {% for scope_handler in oauth.default_scope_handler %}
-            <ScopeHandler class="{{scope_handler}}"/>
-        {% endfor %}
-        {% for scope_handler in oauth.scope_handlers %}
-            <ScopeHandler class="{{scope_handler}}"/>
-        {% endfor %}
+            <ScopeHandler class="org.wso2.carbon.identity.oauth2.validators.OIDCScopeHandler" />
```

Comment:
Add a comment here, why we are not supporting additional handlers

Reply:
Fixed